torch/fx/experimental/proxy_tensor.py

```diff
@@ -512,6 +512,27 @@ def wrapped(flat_args):
         return func(*fn_args, **fn_kwargs)
     return wrapped, flat_args
 
+# Copied from torchdynamo
+# Needed so that tracing in-palce functions doesn't modify inputs
+def clone_input(x):
+    with torch.no_grad():
+        needed_size = sum(
+            (shape - 1) * stride for shape, stride in zip(x.size(), x.stride())
+        )
+        buffer = torch.empty(needed_size + 32, dtype=x.dtype, device=x.device)
+        cache_line_offset = (
+            (x.data_ptr() - buffer.data_ptr()) % 32
+        ) // x.element_size()
```

Comment:
So this is what, necessary because we generate different code depending on the alignment of the allocated tensor?

Reply:
I assume so - I simply copied Jason's code for cloning a tensor.